cmd/swapd/contract.go

```diff
@@ -31,43 +36,29 @@ func getOrDeploySwapFactory(
 	address ethcommon.Address,
 	env common.Environment,
 	dataDir string,
-	privkey *ecdsa.PrivateKey,
-	ec *ethclient.Client,
+	ec extethclient.EthClient,
 	forwarderAddress ethcommon.Address,
-) (*contracts.SwapFactory, ethcommon.Address, error) {
-	var (
-		sf  *contracts.SwapFactory
-		err error
-	)
+) (ethcommon.Address, error) {
+	var err error
 
 	if env != common.Mainnet && (address == ethcommon.Address{}) {
 		// we're on a development or testnet environment and we have no deployed contract,
 		// so let's deploy one
-		address, sf, err = deploySwapFactory(ctx, ec, privkey, forwarderAddress, dataDir)
+		address, _, err = deploySwapFactory(ctx, ec.Raw(), ec.PrivateKey(), forwarderAddress, dataDir)
 		if err != nil {
-			return nil, ethcommon.Address{}, fmt.Errorf("failed to deploy swap factory: %w", err)
+			return ethcommon.Address{}, fmt.Errorf("failed to deploy swap factory: %w", err)
 		}
 	} else {
 		// otherwise, load the contract from the given address
 		// and check that its bytecode is valid (ie. matches the
 		// bytecode of this repo's swap contract)
-		sf, err = getSwapFactory(ec, address)
+		_, err = contracts.CheckSwapFactoryContractCode(ctx, ec.Raw(), address)
 		if err != nil {
-			return nil, ethcommon.Address{}, err
-		}
-		log.Infof("loaded SwapFactory.sol from address %s", address)
```

Comment:
`sf` wasn't being used by any of the callers, so I removed it from the returned values. The print probably doesn't matter, since the user passed the address on the command line.